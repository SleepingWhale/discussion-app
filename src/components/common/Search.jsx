import React from 'react';
import PropTypes from 'prop-types';
import styles from './Search.module.css';

export function Search({ value, onChange }) {
  return (
    <input
      type="search"
      className={styles.input}
      placeholder="Search"
      value={value}
      onChange={onChange}
    />
  );
}

Search.propTypes = {
  value: PropTypes.string,
  onChange: PropTypes.func.isRequired
};

Search.defaultProps = {
  value: ''
};
