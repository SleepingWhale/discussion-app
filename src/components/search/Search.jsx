import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from './Search.module.css';

export class Search extends Component {
  static propTypes = {
    value: PropTypes.string,
    onChange: PropTypes.func.isRequired
  };

  static defaultProps = {
    value: ''
  };

  onChange = e => {
    const { onChange } = this.props;
    onChange(e.target.value);
  };

  handleClear = () => {
    const { onChange } = this.props;
    onChange('');
  };

  render() {
    const { value } = this.props;
    const shouldShowClear = value.length > 0;

    return (
      <div className={styles.container}>
        <input
          type="search"
          className={styles.input}
          placeholder="Search"
          value={value}
          onChange={this.onChange}
          ref={this.inputRef}
        />
        {shouldShowClear && (
          <button
            type="button"
            className={styles.clearBtn}
            onClick={this.handleClear}
          >
            <i className="material-icons md-18">cancel</i>
          </button>
        )}
      </div>
    );
  }
}
