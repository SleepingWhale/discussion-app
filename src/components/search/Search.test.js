import React from 'react';
import { shallow, mount } from 'enzyme';
import { Search } from './Search';

describe('Search', () => {
  it('should render correctly', () => {
    const onChangeHandler = jest.fn();
    const component = shallow(
      <Search onChange={onChangeHandler} value="text" />
    );
    expect(component).toMatchSnapshot();
  });

  it('should clear input', () => {
    const onChangeHandler = jest.fn();
    const content = 'hello world';
    const component = mount(<Search onChange={onChangeHandler} value="text" />);
    const input = component.find('input').at(0);

    input.instance().value = content;
    input.simulate('change');

    expect(onChangeHandler).toHaveBeenNthCalledWith(1, content);

    const clearBtn = component.find('button');
    clearBtn.simulate('click');

    expect(onChangeHandler).toHaveBeenNthCalledWith(2, '');
  });
});
