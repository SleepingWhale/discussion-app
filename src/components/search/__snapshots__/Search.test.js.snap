// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render correctly 1`] = `
<div
  className="container"
>
  <input
    className="input"
    onChange={[Function]}
    placeholder="Search"
    type="search"
    value="text"
  />
  <button
    className="clearBtn"
    onClick={[Function]}
    type="button"
  >
    <i
      className="material-icons md-18"
    >
      cancel
    </i>
  </button>
</div>
`;
